app/firebase/functions/src/notifications/notification.task.ts

```diff
@@ -0,0 +1,48 @@
+import { Request } from 'firebase-functions/v2/tasks';
+
+import {
+  ActivityEvent,
+  NOTIFICATION_FREQUENCY,
+} from '../@shared/types/types.notifications';
+import { logger } from '../instances/logger';
+import { createServices } from '../instances/services';
+
+export const SEND_NOTIFICATION_TASK = 'sendNotification';
+
+export const sendNotificationTask = async (req: Request) => {
+  logger.debug(
+    `sendNotificationTask: activityEventId: ${req.data.activityEvent}`
+  );
+  const activityEvent = req.data.activityEvent as ActivityEvent;
```

Comment:
this is a WIP, probably want to switch this out for a `Notification` object instead